sources/Core/Image.cpp

```diff
@@ -70,7 +70,7 @@ Image& Image::operator = (const Image& rhs)
     return *this;
 }
 
-Image& Image::operator = (Image&& rhs)
+Image& Image::operator = (Image&& rhs) noexcept
 {
     Reset(rhs.GetExtent(), rhs.GetFormat(), rhs.GetDataType(), std::move(rhs.data_));
```

Comment:
C++ doesn't provide compile-time checks for `noexcept` functions, but as I understand it, any functions being called inside a `noexcept`-declared function _should_ also guarantee to never throw an exception. So that means `Reset()`, `ResetAttributes()`, `GetFormat()`, and `GetDataType()` should also have `noexcept`, which is fine for all other than `Reset()` IMHO. Functions that call `delete` don't look like good `noexcept`-candidates to me, but maybe I'm missing something. I can find arguments for and [against it](https://news.ycombinator.com/item?id=41163083).